lms/djangoapps/course_api/blocks/forms.py

```diff
@@ -106,30 +107,70 @@ def clean(self):
         cleaned_data['user'] = self._clean_requested_user(cleaned_data, usage_key.course_key)
         return cleaned_data
 
+    def clean_username(self):
+        """
+        Return cleaned username.
+
+        Overrides the default behaviour that maps an empty string to None. This
+        allows us to differentiate between no username being provided (None) vs
+        an empty username being provided ('').
+        """
+        if 'username' in self.data:
+            return self.cleaned_data['username']
+        else:
```

Comment:
If an `if` block ends with a `return` statement, the `else` is superfluous and we can remove the `else`, unindent its body, and make it a little easier to read.